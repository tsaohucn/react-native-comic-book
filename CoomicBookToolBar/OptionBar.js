import React from 'react'
import { 
  View, 
  StyleSheet, 
  Text, 
  Animated, 
  Dimensions, 
  TouchableWithoutFeedback ,
  FlatList
} from 'react-native'
import { Icon } from 'react-native-elements'

const OptionBar = ({animatedOptionBarY}) => (
    <Animated.View style={[styles.optionBarAnimated,{
      transform: [
        { translateY: animatedOptionBarY }
      ]}]}
    >
      <View style={styles.optionBarRaw}>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='arrow-down-bold-circle-outline'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>下載</Text> 
          </View> 
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='content-copy'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>詳情</Text> 
          </View> 
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='comment-processing-outline'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>評論</Text> 
          </View> 
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='share'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>分享</Text> 
          </View> 
        </TouchableWithoutFeedback>
      </View>
      <View style={styles.optionBarRaw}>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='star-circle'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>收藏</Text> 
          </View> 
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='ticket'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>投月票</Text> 
          </View> 
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          //onPress={} 
         >
          <View style={styles.iconView}>
            <Icon
                name='bookmark'
                type='material-community'
                color='white'
              />
              <Text style={styles.iconText}>標籤</Text> 
          </View> 
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback>
          <View style={styles.iconView}>
          </View> 
        </TouchableWithoutFeedback>
      </View>
    </Animated.View>
)

const { width, height } = Dimensions.get('window')

const styles = StyleSheet.create({
  iconText: {
    color: 'white'
  },
  optionBarAnimated: {
    position: 'absolute', 
    height: 100,
    width,
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
    bottom: 0     
  },
  optionBarRaw: {
    flex: 1,
    flexDirection: 'row'
  },
  iconView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
})

export default OptionBar