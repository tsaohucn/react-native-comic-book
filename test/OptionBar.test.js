// __tests__/OptionBar.js
import 'react-native'
import React from 'react'
import OptionBar from '../CoomicBookTool/OptionBar'

import renderer from 'react-test-renderer'

jest.mock('react-native-elements', () => {
	return {
		Icon: 'Icon'
	}
})

describe('Testing OptionBar component', () => {

	it('renders correctly with no props', () => {
		const props = {

		}
	  const tree = renderer.create(
	    <OptionBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

	it('renders correctly with animatedOptionBarY value', () => {
		const props = {
			animatedOptionBarY: 100
		}
	  const tree = renderer.create(
	    <OptionBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

})