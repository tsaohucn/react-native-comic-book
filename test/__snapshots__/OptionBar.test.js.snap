// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing OptionBar component renders correctly with animatedOptionBarY value 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "bottom": 0,
      "height": 100,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 100,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="arrow-down-bold-circle-outline"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        下載
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="content-copy"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        詳情
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="comment-processing-outline"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        評論
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="share"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        分享
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="star-circle"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        收藏
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="ticket"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        投月票
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="bookmark"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        標籤
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    />
  </View>
</View>
`;

exports[`Testing OptionBar component renders correctly with no props 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "bottom": 0,
      "height": 100,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": undefined,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="arrow-down-bold-circle-outline"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        下載
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="content-copy"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        詳情
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="comment-processing-outline"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        評論
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="share"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        分享
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="star-circle"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        收藏
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="ticket"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        投月票
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Icon
        color="white"
        name="bookmark"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        標籤
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    />
  </View>
</View>
`;
